OpenRA.Mods.Common/Traits/Render/RenderDebugState.cs

```diff
@@ -81,8 +82,26 @@ IEnumerable<IRenderable> IRenderAboveShroudWhenSelected.RenderAboveShroud(Actor
 			var activity = self.CurrentActivity;
 			if (activity != null)
 			{
-				var activityName = activity.GetType().ToString().Split('.').Last();
-				yield return new TextRenderable(font, self.CenterPosition, 0, color, activityName);
+				var act = activity;
+				List<Activity> activities = new List<Activity>();
+				while (act != null)
+				{
+					activities.Add(act);
+					act = act.CurrentChildActivity;
+				}
+
+				var activityString = "";
```

Comment:
This should be able to simplify down to something like `activityString = activities.Select(a => a.GetType().ToString().Split('.').Last()).JoinWith(".");`

Reply:
Or even simpler
var activityString = activities.Select(a => a.GetType().Name).JoinWith(".");